### PR TITLE
Workaround intel 19.1.x compiler issue

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Expect to fix https://github.com/QMCPACK/qmcpack/issues/3955.
Intel compiler was choked by `std::vector::insert(std::variant<>)`

## What type(s) of changes does this code introduce?
- Bugfix

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
Theta

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by ye-luo** [comment]
Test this please

File: src/Estimators/EstimatorManagerInput.cpp [hunk 1]
```diff
@@ -22,11 +22,28 @@ EstimatorManagerInput::EstimatorManagerInput(xmlNodePtr cur) { readXML(cur); }
 
 EstimatorManagerInput::EstimatorManagerInput(std::initializer_list<EstimatorManagerInput> emil)
 {
-  std::for_each(emil.begin(), emil.end(), [this](const EstimatorManagerInput& emi) {
-    std::copy(emi.estimator_inputs_.begin(), emi.estimator_inputs_.end(), std::back_inserter(estimator_inputs_));
+  // \todo the following code fusing two vectors can be written in more consice way with std::back_inserter. See history.
+  // Right now needs to use a clumsy way to make intel classic compiler 19.1.x happy when using gcc 9 on some OS distros.
```
Comment (ye-luo) [inline src/Estimators/EstimatorManagerInput.cpp:26]: @PDoakORNL Added TODO note here.
Q: does std::back_inserter potentially over allocate memory?
I'm not saying memory is a concern here. Just to understand how back_inserter works.

**Comment by ye-luo** [comment]
Test this please

File: src/Estimators/EstimatorManagerInput.cpp (hunk 1 above)
Comment (prckent) [reply src/Estimators/EstimatorManagerInput.cpp:26]: If there are concerns we can put an #ifdef for this particularly compiler, just as we have in the past. There are still alot of Intel 19.x compilers in use, so if we can make things easier for users easily we should do so.

Comment (PDoakORNL) [reply src/Estimators/EstimatorManagerInput.cpp:26]: Considering the workaround is localized the think the \todo is less trouble.

Comment (PDoakORNL) [reply src/Estimators/EstimatorManagerInput.cpp:26]: And back_inserter works like the containers push_back.

**Review by PDoakORNL — APPROVED** [review]
So I guess this works in later intel + stdlib combinations?

**Pull request merged**